File: mianyang-campus/backend/app/schemas/service.py
from pydantic import BaseModel
from datetime import datetime
from typing import Any


class TicketCreate(BaseModel):
    type: str
    title: str
    content: str = ""
    applicant_name: str = ""
    applicant_no: str = ""
    applicant_college: str = ""
    form_data: dict[str, Any] | None = None
    attachments: list[str] | None = None


class TicketOut(BaseModel):
    id: int
    applicant_id: int
    applicant_name: str
    applicant_no: str
    applicant_college: str
    type: str
    title: str
    content: str
    form_data: dict[str, Any] | None = None
    attachments: list[str] | None = None
    status: str
    created_at: datetime

    class Config:
        from_attributes = True


class TicketApprove(BaseModel):
    action: str
    comment: str | None = None

File: mianyang-campus/frontend/src/api/service.ts
import request from '@/utils/request'
import type { ServiceTicket } from '@/types'

export function getTickets(params?: { status?: string }) {
  return request.get<ServiceTicket[]>('/service/tickets', { params })
}

export function createTicket(data: {
  type: string; title: string; content?: string
  applicant_name?: string; applicant_no?: string; applicant_college?: string
  form_data?: Record<string, any>; attachments?: string[]
}) {
  return request.post<ServiceTicket>('/service/tickets', data)
}

export function approveTicket(id: number, action: string) {
  return request.put(`/service/tickets/${id}/approve`, { action })
}

export function cancelTicket(id: number) {
  return request.put(`/service/tickets/${id}/cancel`)
}